qCC/ccComparisonDlg.cpp

```diff
@@ -940,6 +947,28 @@ bool ccComparisonDlg::computeDistances()
 				}
 			}
 			ccLog::Warning("[ComputeDistances] Result has been split along each dimension (check the 3 other scalar fields with '_X', '_Y' and '_Z' suffix!)");
+            if (mergeXY)
+            {
+                ccLog::Warning("[ComputeDistances] compute 2D distance (xy plane)");
+                int sf2D = m_compCloud->getScalarFieldIndexByName(qPrintable(m_sfName + QString(" (XY)")));
+                if (sf2D >= 0)
+                    m_compCloud->deleteScalarField(sf2D);
+                sf2D = m_compCloud->addScalarField(qPrintable(m_sfName + QString(" (XY)")));
+                CCCoreLib::ScalarField* sf = m_compCloud->getScalarField(sf2D);
+                for (int idx = 0; idx < m_compCloud->size(); idx++)
+                {
+                    float d2D = pow(pow(c2cParams.splitDistances[0]->getValue(idx), 2) +  pow(c2cParams.splitDistances[1]->getValue(idx), 2), 0.5);
+                    sf->setValue(idx, d2D);
+                }
+                sf->computeMinAndMax();
+                // remove X and Y SF
+                int sfX = m_compCloud->getScalarFieldIndexByName(qPrintable(m_sfName + QString(" (X)")));
+                if (sfX >= 0)
+                    m_compCloud->deleteScalarField(sfX);
```

Comment:
same comment as above

Reply:
By principle, I wouldn't remove an existing scalar field silently (and the user can delete it manually).

Reply:
OK